### DISCUSSION TITLE
Exporting via Web App: Add a "Export as displayed" checkmark to the export drawer in the UI

### DISCUSSION THREAD

**Discussion by shadowcpy** [directus/directus#6753] [Feature Requests]
Hi everyone,
this is a follow-up on the discussion #4639.
I would propose adding a checkmark to the Web UI similar to the already existing "Use current filters" checkbox,
which is called "Export as displayed". This setting will influence the exported data when clicking on the export button.
Currently, when an end user views a certain collection with certain settings, and the collection contains related fields,
the "Display Format" setting of the field is used to render the related values on the UI. But once the end user clicks export with the current settings and expects the same output in a csv for example, the file will contain different data.
Instead of applying the Display Format setting to render the related value, the related value will just be exported "as-is" without resolving. This has lead to confusion on some projects as an end-user does not want IDs of related fields but instead expects the  already visible output in the App to be exported to a universal format.

Example:
An accountant uses the Web-App to display all of the customers of a company. Each customer has a employee assigned.
In the App the Name of the Assigned Employee is shown. But when the accountant wants to export the data as a csv, for example to print it, instead of the names of the employees, there are just the IDs from the related column.

When checking this box, the same logic that is already used to render the collection (the Display Format field setting) should be used to render the exported related data. 

This approach should be less of a big breaking change than the already planned Virtual Columns while resolving the most common issue with related field exporting.

Please let me know of any improvements or ideas,
m0rtis

**Comment by benhaynes** [comment]
I know there was a lot of talk about virtual fields in that other ticket, but another option (in addition) is:

Using a tree view for selecting deeply nested fields from within the Collection's settings. This would provide the default fields for layouts and even the API. Not sure how this would be customized _within_ the layouts, though.

Am I correct with this, @rijkvanzanten ?

**Reply by rijkvanzanten** [reply]
It might not be a bad idea to make export a more advanced dialog where you get to choose the exact fields you want to export, instead of relying on what the layout might display or not. This also makes sure we can always present the export functionality regardless of layout used.

**Reply by shadowcpy** [reply]
+1

**Comment by shadowcpy** [comment]
Hey everyone,
any progress on this? 
I tried to start implementing it myself, but even though I hate to admit, the code that is used to format the relational values is too complex for me to understand, especially in https://github.com/directus/directus/blob/main/app/src/utils/render-string-template.ts
and https://github.com/directus/directus/blob/main/app/src/displays/related-values/related-values.vue
Though I understand that the display is used for rendering the Relational values and therefore the additions to support this feature would probably need to be made in related-values.vue I don't think I will be able to implement it.
WIll look at this again soon and maybe I can make sense of it then

**Reply by rijkvanzanten** [reply]
The export happens all API side by adding the `export` query parameter. Therefore, the only thing the app should do is to make sure the endpoint is called with the correct query parameters (next to `export`), like `fields`, `filter`, etc.

This makes this particular feature request super difficult though, as the displayed value is handled by the _app_, not the API. So in order to export what you _see_ instead of export the data, we'll have to add a secondary export flow to render things into a data string, and save that to the browser somehow 🤔

**Reply by shadowcpy** [reply]
Oh I just realized it too... I think the problem here is that the api does not have access to all the fancy related resolving and translations, as api and app are strictly separated. This would effectively mean moving the necessary functionality out of the app or reimplementing it in the API, though. I am not sure how to proceed with this...

**Reply by rijkvanzanten** [reply]
The API can do the relational fetching, but it doesn't do any string template rendering like the app does, so yeah you're right. we'll either have to re-implement the APIs creation of a file in the app, or recreate the apps rendering of the display template in the API. Neither are easy nor ideal..

**Reply by shadowcpy** [reply]
> The API can do the relational fetching, but it doesn't do any string template rendering like the app does, so yeah you're right. we'll either have to re-implement the APIs creation of a file in the app, or recreate the apps rendering of the display template in the API. Neither are easy nor ideal..

Yeah, agreed. But I think that creating the csv file using the data from the state of the app (collections page) is maybe better than having to basically recreate the whole Displays architecture for the api....

**Reply by rijkvanzanten** [reply]
Hmm yeah, but that has the downside that it'll never work for datasets that are larger than what the app can display 🤔

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
